### ISSUE TITLE
Editor: Replace block editor back icon button with icon and text button

### ISSUE THREAD

**Issue by apeatling** [Automattic/wp-calypso#37284]
See the last three minutes of the "Making First Home Page Edits" testing video on this post: p58i-8d4-p2

* When prompted to look at the “<” back button in the top left he realizes he can leave the editor and go back to Calypso to do other things like change the design. “I totally forgot about this page!”.

------

Users are not seeing the simple "<" as any indication that they can go back to the page behind. We need to be much clearer with this button and include a call to action depending on where they are going to go.

**Comment by noahtallen** [comment]
Conveniently enough, this will resolve this issue!

https://github.com/Automattic/wp-calypso/pull/37234

**Comment by noahtallen** [comment]
cc @Addison-Stavlo

**Comment by noahtallen** [comment]
(unless there is more to do here.) Currently, this will look like the following:

**Page Editor**
<img width="1919" alt="Screen Shot 2019-11-04 at 12 11 27 PM" src="https://user-images.githubusercontent.com/6265975/68154397-4c7d3b00-fefc-11e9-8c21-822227d78ac3.png">


**Template Editor**
<img width="1774" alt="Screen Shot 2019-11-04 at 12 12 09 PM" src="https://user-images.githubusercontent.com/6265975/68154430-5f900b00-fefc-11e9-8f4f-c09520f9028b.png">

**Comment by shaunandrews** [comment]
This seems like an issue best served in Core.

**Comment by noahtallen** [comment]
Yeah, building extension points for the back button is a great idea. IMO, we should add settings like:

- Label overrides
- Always display, even on small viewports
- Href overrides.

These would all make life easier for us.

**Comment by Addison-Stavlo** [comment]
I see @apeatling assigned me to this task two hours ago. 

I am a little confused.  We did have this change set up in #37234 but ended up reverting it per design request.  I can make this happen quickly in our override in the FSE plugin, but there has also been discussion that this should be handled in core.  

Being that this is issue is being assigned via wp-calipso repo and is labeled 'high priority', should I go ahead and do the quick fix in our plugin first and look to core later?  Or should I be carrying this all out via core contributions?

If for the quick fix in wp-calypso repo, should I make this happen for both the page and posts editor? ( It was previously mentioned that I should leave the posts editor alone) .  Also, do we change the wp_template_part editor to match this format as well or leave it as the blue secondary button that it currently is as @shaunandrews suggested?

Thanks!

**Comment by noahtallen** [comment]
To speak towards our "process":

We typically use assignments to note who is working on something or who has been working on something, not to indicate that you should change focus. In general, you can even feel free to assign yourself or others to issues you think are relevant to them. And you can also see if someone else can take over if you think they'd be a better fit. :)

I don't think there's any urgency in this (i.e. don't stay up late over it). The high priority label would just mean that it's something that should be worked on before lower priority items, but doesn't mean we should rush a fix. I think it's worth investigating core changes to resolve this issue. In my very limited experience here, when we delay fixing something for real (i.e. "handle it in a follow up"), we usually don't handle it at all.

And to clarify what core changes could look like, just adding settings and extension points such that we could change the behavior of the back button for WordPress.com. We probably wouldn't change the behavior globally for all of Gutenberg. For example, we could add an editor setting for always showing the back button, even on small viewports, and then we can use that setting for WordPress.com.

**Comment by apeatling** [comment]
@Addison-Stavlo I assigned you as per this comment: https://github.com/Automattic/wp-calypso/pull/37234#issuecomment-550541005

No immediate urgency here, @noahtallen's comment covers it well, thanks Noah.

**Comment by noahtallen** [comment]
cc @shaunandrews @olaolusoga @apeatling @vindl 

After lengthy discussion over at https://github.com/WordPress/gutenberg/pull/18387, the general consensus is that we shouldn't add extension points to the back button. Rather, the suggestion is that we should build our own editor out of the pieces given to us in edit-post.

Since we have explored that before and decided that it really wouldn't be practical, I imagine the "best" solution here is just to continue with the hacks for the time being.

That being said, I do think we will increasingly be forced to examine whether or not building our own editor makes sense, and it may be worth another quick exploration. To clarify, I mean replacing the editor on the wp-admin level, not bundling it in Calypso.

**Comment by vindl** [comment]
> Since we have explored that before and decided that it really wouldn't be practical, I imagine the "best" solution here is just to continue with the hacks for the time being.

I concur. 

> That being said, I do think we will increasingly be forced to examine whether or not building our own editor makes sense, and it may be worth another quick exploration. To clarify, I mean replacing the editor on the wp-admin level, not bundling it in Calypso.

This would be great if `edit-post` [exported](https://github.com/WordPress/gutenberg/blob/994a9f25386bfd5a3a74bfb537e36684b058b286/packages/edit-post/src/index.js) all of the existing components in the tree that we want to reuse and leave unchanged. But since that's still not the case we'd end up having to duplicate pretty much everything there just for the sake of being able to customize ~1% of the UI that we need. Until that is resolved this option seems prohibitive to me from a maintenance point of view. We tried this exact same thing during the Gutenlypso era and it was a mess.

**Comment by apeatling** [comment]
Where does the above leave this issue? What do we need to do next?

**Comment by noahtallen** [comment]
Basically, continue with the approach @Addison-Stavlo started in the PR that we put on pause: use the same back button component everywhere and add the text mentioned in the issue here. Should be pretty quick since we aren’t blocked on core :) IMO we should do this for all post types, including the normal post editor.

**Comment by Addison-Stavlo** [comment]
It seems like the core discussion was starting to agree with the idea of adding the visible label to the button.  But not with the idea of ever rendering the button in wp-admin mode, and also made it seem that adding the button for mobile screen widths would be a large task over there.

So did we want to go back to doing everything on our end through dotcom overrides?  Or were we still looking to try to get some of these pieces done over on core?

**Comment by noahtallen** [comment]
I think it's worth continuing the discussion in that PR about showing text, but I'm still worried we (the core gutenberg team) wouldn't want to add a way to override what the text is. So my vote would be for continuing the discussion, but implementing the dotcom overrides

**Comment by apeatling** [comment]
Agreed, let's continue the core discussion but in the meantime implement the dotcom overrides.
